pokedex_async_redux/lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -82,4 +155,23 @@ class PokemonOverviewPage extends StatelessWidget {
       MaterialPageRoute(builder: (_) => PokemonDetailsConnector(pokemon: pokemon)),
     );
   }
+
+  void _clearSearchedPokemons() {
+    widget.clearSearchedPokemons;
+    setState(() {
+      _searchController.text = emptyString;
+      isSearching = false;
+    });
+  }
+
+  void _duringPokemonSearch(String queryText) {
+    print('_duringPokemonSearch queryText: $queryText');
+    _debounceTimer = Timer(
```

Comment:
seems you need to have this validation above this line to make sure there's only one active debouncer running 🙂 

`if (_debounce?.isActive ?? false) _debounce?.cancel();`